nomenclature/processor/region.py

```diff
@@ -349,6 +349,21 @@ def apply(self, df: IamDataFrame, dsd: DataStructureDefinition) -> IamDataFrame:
                         }
                         for cr in self.mappings[model].common_regions:
                             regions = [cr.name, cr.constituent_regions]
+                            # Check for already present variables
+                            model_native_res = model_df.filter(region=cr.name)
+                            if len(model_native_res):
+                                processed_dfs.append(model_native_res)
+                                model_native_vars = set(model_native_res.variable)
+                                # Remove all already present variables
+                                vars_default_args = list(
```

Comment:
I think that you have to create a copy here...
If you perform this operation for several common regions, you remove arguments from the dictionary in each iteration - so if the first region has *Variable A* (and is omitted from the aggregation), that variable will also be ignored for the second region...

Reply:
Ah yes, good catch, the copy will be necessary. 
I'll also create a dedicated, more focused test case that covers multiple common regions.

Reply:
done.